solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -970,27 +1028,33 @@ protected void mergeIds(ResponseBuilder rb, ShardRequest sreq) {
             }
           }
 
+          shardDoc.rawSortFieldValues = sortFieldValues;
```

Comment:
I can update to only set this if `rb.isIterative()`. Potentially hold onto less memory in the non iterative case.